packages/medusa/src/services/cart.ts

```diff
@@ -1213,6 +1213,13 @@ class CartService extends BaseService {
         return cartRepository.save(cart)
       }
 
+      if (!cart.payment_session) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          "You cannot complete a cart without a payment session."
+        )
+      }
```

Comment:
nit: While we are at it, should we also enforce shipping methods for cart completion to go through?